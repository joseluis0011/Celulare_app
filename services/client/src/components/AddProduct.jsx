import React from 'react';
const AddProduct= (props) => {
    return (
    <form onSubmit={(event) => props.addProduct(event)}>
        <div className="field">
        <div className="control">
        <input
        name="nombre"
        className="input is-large"
        type="text"
        placeholder="celular"
        required
        value={props.nombre}
        onChange={props.handleChange}
        />
        </div>
        </div>
        <div className="field">
        <input
        name="cantidad"
        className="input is-large"
        type="text"
        placeholder="cantidad"
        required
        value={props.cantidad}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="serie"
        className="input is-large"
        type="text"
        placeholder="serie"
        required
        value={props.serie}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="modelo"
        className="input is-large"
        type="text"
        placeholder="modelo"
        required
        value={props.modelo}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        name="marca"

        className="input is-large"
        type="text"
        placeholder="marca"
        required
        value={props.marca}
        onChange={props.handleChange}
        />
        </div>
        <div className="field">
        <input
        type="submit"
        className="button is-primary is-large is-rounded is-fullwidth"
        value="Registrar"
        />
        </div>
        </form>
        )
    };

export default AddProduct;