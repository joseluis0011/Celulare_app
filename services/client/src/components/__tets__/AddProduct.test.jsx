import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddProduct from '../AddProduct';

test('AddProduct renders properly', () => {
    const wrapper = shallow(<AddProduct/>);
    const element = wrapper.find('form');
    expect(element.find('input').length).toBe(6);
    expect(element.find('input').get(0).props.name).toBe('nombre');
    expect(element.find('input').get(1).props.name).toBe('cantidad');
    expect(element.find('input').get(2).props.name).toBe('serie');
    expect(element.find('input').get(3).props.name).toBe('modelo');
    expect(element.find('input').get(4).props.name).toBe('marca');
    expect(element.find('input').get(5).props.type).toBe('submit');
});

test('AddProduct renders a snapshot properly', () => {
    const tree = renderer.create(<AddProduct/>).toJSON();
    expect(tree).toMatchSnapshot();
});
