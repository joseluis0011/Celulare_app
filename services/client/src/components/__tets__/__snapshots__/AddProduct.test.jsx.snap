// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProduct renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <input
        className="input is-large"
        name="nombre"
        placeholder="celular"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="cantidad"
      placeholder="cantidad"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="serie"
      placeholder="serie"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="modelo"
      placeholder="modelo"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="marca"
      placeholder="marca"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="button is-primary is-large is-rounded is-fullwidth"
      type="submit"
      value="Registrar"
    />
  </div>
</form>
`;
